src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs

```diff
@@ -256,5 +247,21 @@ private static bool SupportsPdbGeneration()
 
             return false;
         }
+
+        private static string ReadFileContentsSafely(IFileInfo fileInfo)
```

Comment:
It seems super trivial to be shared. I'd rather it be duped than have it be oddly misplaced on the error class.